### PR TITLE
feat: Auto update chectl if user confirms the update

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
When a user tries to install or update Eclipse Che and chectl version is not the latest one, we need to warn the user that in order to have the latest Che, chectl must be updated. If user confirms the update, then chectl will update itself.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19128

### How to test this PR?
Run `chectl server:update` command when there is an update available.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -162,12 +163,12 @@ export function notifyCommandCompletedSuccessfully(): void {
...
 
-export async function askForChectlUpdateIfNeeded(): Promise<void> {
+export async function askForChectlUpdateIfNeeded(command: Command): Promise<void> {
   const ctx = ChectlContext.get()
   if (await VersionHelper.isChectlUpdateAvailable(ctx[ChectlContext.CACHE_DIR])) {
-    cli.info('A newer version of chectl is available.')
-    if (await cli.confirm('To deploy the latest version of Eclipse Che you have to update chectl first [y/n]')) {
-      cli.info('Please run "chectl update" and then repeat "server:deploy" command.')
+    cli.info('A newer version of chectl is available. To deploy the latest version of Eclipse Che you have to manually update chectl first.')
+    if (!await cli.confirm('Do you want to continue using older version now ? [y/n]')) {
```
Comment (beaumorley) [inline src/util.ts:170]: This seems backwards. What happens if they choose y/n here? What about flipping it.  "Do you want to update chectl now  [y/n]" and if they say y then give them the command to do so. If they say no then open the old version.

**Review by AndrienkoAleksandr — APPROVED** [review]

File: src/util.ts [hunk 2]
```diff
@@ -162,12 +163,12 @@ export function notifyCommandCompletedSuccessfully(): void {
...
-export async function askForChectlUpdateIfNeeded(): Promise<void> {
+export async function askForChectlUpdateIfNeeded(command: Command): Promise<void> {
   const ctx = ChectlContext.get()
   if (await VersionHelper.isChectlUpdateAvailable(ctx[ChectlContext.CACHE_DIR])) {
-    cli.info('A newer version of chectl is available.')
-    if (await cli.confirm('To deploy the latest version of Eclipse Che you have to update chectl first [y/n]')) {
-      cli.info('Please run "chectl update" and then repeat "server:deploy" command.')
+    cli.info('A newer version of chectl is available. To deploy the latest version of Eclipse Che you have to manually update chectl first.')
+    if (await cli.confirm('Do you want to update chectl now [y/n]')) {
+      cli.info(`Please run "chectl update" and then repeat "${command.id}" command.`)
```
Comment (tolusha) [inline src/util.ts:171]:
```suggestion
      cli.info(`Please update chectl and then repeat command again:`)
      cli.info(`1. chectl update`)
      cli.info(`2. chectl ${command.id} ...`)
```

File: src/util.ts [hunk 3]
```diff
@@ -162,12 +163,14 @@ export function notifyCommandCompletedSuccessfully(): void {
...
-export async function askForChectlUpdateIfNeeded(): Promise<void> {
+export async function askForChectlUpdateIfNeeded(command: Command): Promise<void> {
   const ctx = ChectlContext.get()
   if (await VersionHelper.isChectlUpdateAvailable(ctx[ChectlContext.CACHE_DIR])) {
-    cli.info('A newer version of chectl is available.')
-    if (await cli.confirm('To deploy the latest version of Eclipse Che you have to update chectl first [y/n]')) {
-      cli.info('Please run "chectl update" and then repeat "server:deploy" command.')
+    cli.info('A newer version of chectl is available. To deploy the latest version of Eclipse Che you have to manually update chectl first.')
+    if (await cli.confirm('Do you want to update chectl now [y/n]')) {
+      cli.info('lease update chectl and then repeat command again:')
```
Comment (tolusha) [inline src/util.ts:171]:
```suggestion
      cli.info('Please update chectl and then repeat command again:')
```

File: src/util.ts [hunk 4]
```diff
@@ -162,12 +163,14 @@ export function notifyCommandCompletedSuccessfully(): void {
...
 }
 
-export async function askForChectlUpdateIfNeeded(): Promise<void> {
+export async function askForChectlUpdateIfNeeded(command: Command): Promise<void> {
   const ctx = ChectlContext.get()
   if (await VersionHelper.isChectlUpdateAvailable(ctx[ChectlContext.CACHE_DIR])) {
-    cli.info('A newer version of chectl is available.')
-    if (await cli.confirm('To deploy the latest version of Eclipse Che you have to update chectl first [y/n]')) {
-      cli.info('Please run "chectl update" and then repeat "server:deploy" command.')
+    cli.info('A newer version of chectl is available. To deploy the latest version of Eclipse Che you have to manually update chectl first.')
```
Comment (tolusha) [inline src/util.ts:169]:
```suggestion
    cli.info('A more recent version of chectl is available. To deploy the latest version of Eclipse Che, update the chectl tool first.')
```

File: src/util.ts [hunk 5]
```diff
@@ -162,12 +166,21 @@ export function notifyCommandCompletedSuccessfully(): void {
...
-      cli.info('Please run "chectl update" and then repeat "server:deploy" command.')
+    cli.info('A newer version of chectl is available. To deploy the latest version of Eclipse Che you have to update chectl first.')
+    if (await cli.confirm('Do you want to update chectl now? [y/n]')) {
+      // Update chectl
+      await UpdateCommand.run([])
+
+      // Rerun the command
+      if (command.id === 'server:update') {
+        await Update.run(command.argv)
+      } else {
```
Comment (tolusha) [inline src/util.ts:180]:
```suggestion
      } else if (command.id === 'server:deploy') {
```

**Review by tolusha — APPROVED** [review]

**Review by tolusha — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Comment by mmorhun** [comment]
Rebased

**Pull request merged**